### ISSUE TITLE
allow for log scale in `RangeFilterState` range selection plot?

### ISSUE THREAD

**Issue by chlebowa** [insightsengineering/teal.slice#321]
### Feature description

Cases like this

![image](https://github.com/insightsengineering/teal.slice/assets/114988527/b905d51b-7471-44ac-b03b-e45ca06bbea8)

could benefit from presenting the x axis in logarithmic scale, like this:

![image](https://github.com/insightsengineering/teal.slice/assets/114988527/23942b53-7acb-490d-b73d-0f3d9bf19c79)

Would users be interested in such customization? If so, we could also allow for modifying bin number in there.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

**Comment by lcd2yyz** [comment]
Great suggestion! I definitely think there would be users that prefer log visualization in these situations, although we'd need to think through the detailed behavior more robustly. 
- I would only want the distribution plot as the visual aid to be shown as log scale, with the x-label shown as `1, 10, 10^2, 10^3` instead of `0, 1, 2, 3`. User would still use the normal scale for numerical input - otherwise interpretability will be difficult for non-data orientated users. 
- What if the original value is 0? How should the count for 0 be displayed?
- Is this something that can only be configured by the app developer? Or should it be offered as some UI toggle to switch between normal vs log views?

**Comment by m7pr** [comment]
Just pointing this out that we need to use log(1+x) transformation instead of log(x) so that we can also handle the distribution of values that are non-negative.

**Comment by lcd2yyz** [comment]
Another idea is instead of applying log-transformation, use square-root (or even cube root) transformation instead? This will help to circumvent the issue with non-positive numbers. So if we can sensibly label the x-axis with linear scale numbers, it should still offer reasonable interpretability.

**Comment by m7pr** [comment]
@lcd2yyz square root has the same limitation as log function, they both work only for non-negative values (square root also works for zero), but they both do not support negative values. However your message made me thinking we should support multiple typical scaling functions as the distribution of the numeric value can be skewed at both ends (not only the left-skewed but also right-skewed). We can also support scaling to a normal distribution, which could be beneficial for statistical models like glm.

**Comment by lcd2yyz** [comment]
> @lcd2yyz square root has the same limitation as log function, they both work only for non-negative values (square root also works for zero), but they both do not support negative values.

Right, so cube root might be the most robust form of transformation (over log and sqrt) when it comes to numerical range slider, as no conditional handling needed. My main concern is still interpretability, in that sense, log > sqrt ~ cbrt (with sqrt slightly better than cbrt).

The more advanced transformation like scaling is useful for preparing data for modeling, but then we are talking about data manipulation or data processing (so maybe should be an option that goes to the LHS encoding panels), rather than data visualization (what we are trying to do with slider for filter panel). It would become impossible to interpret the true data ranges with a scaled distribution.

**Comment by m7pr** [comment]
If there is an option, probably data transformations should have their own panel or should be manipulated by the app creator in the initial code that runs the app

**Comment by lcd2yyz** [comment]
As it stands, enabling a transformed visualization for the `RangeFilterState` is not an required feature for our main filter panel refactor objective, it's an nice-to-have. 
I would suggest to turn this into a research issue, whoever take this on should propose a design (UI and the backend data handling), then bring back to the team and PO for review and feedback.
